source/patterns/@aws-solutions-constructs/aws-events-rule-sqs/test/events-rule-sqs-queue.test.ts

```diff
@@ -0,0 +1,212 @@
+                  "sqs:GetQueueAttributes",
+                  "sqs:GetQueueUrl",
+                ],
+                Effect: "Allow",
+                Principal:  {
+                  "AWS":  {
+                    "Fn::GetAtt": [
+                      "testeventsrulesqsEventsRule06054F3F",
+                      "Arn",
+                    ],
+                  },
+                },
+                Resource:  {
+                  "Fn::GetAtt": [
+                    "testeventsrulesqsqueueAACD0364",
+                    "Arn",
+                  ],
+                },
+              },
+            ],
+            Version: "2012-10-17"
+        },
+            Queues: [
+                {
+                  "Ref": "testeventsrulesqsqueueAACD0364",
+                }
+            ]     
+    })
+})
+
+test('check if the dead letter queue policy is setup', () => {
+    const stack = getStack()
+    deployNewFunc(stack)
+    expect(stack).toHaveResource('AWS::SQS::QueuePolicy', {
+        PolicyDocument:  {
+            Statement: [
+               {
+                Action: [
+                  "sqs:DeleteMessage",
+                  "sqs:ReceiveMessage",
+                  "sqs:SendMessage",
+                  "sqs:GetQueueAttributes",
+                  "sqs:RemovePermission",
+                  "sqs:AddPermission",
+                  "sqs:SetQueueAttributes",
+                ],
+                Effect: "Allow",
+                Principal:  {
+                  AWS:  {
+                    "Fn::Join": [
+                      "",
+                      [
+                        "arn:",
+                         {
+                          "Ref": "AWS::Partition",
+                        },
+                        ":iam::",
+                       {
+                        "Ref": "AWS::AccountId"
+                      },
+                      ":root"
+                      ],
+                    ],
+                  },
+                },
+                Resource:  {
+                  "Fn::GetAtt": [
+                    "testeventsrulesqsdeadLetterQueueA4A15A1C",
+                    "Arn",
+                  ],
+                },
+                Sid: "QueueOwnerOnlyAccess",
+              },
+               {
+                Action: "SQS:*",
+                Condition:  {
+                  "Bool":  {
+                    "aws:SecureTransport": "false",
+                  },
+                },
+                Effect: "Deny",
+                Principal: "*",
+                Resource:  {
+                  "Fn::GetAtt": [
+                    "testeventsrulesqsdeadLetterQueueA4A15A1C",
+                    "Arn",
+                  ],
+                },
+                Sid: "HttpsOnly",
+              },
+            ],
+            Version: "2012-10-17",
+          },
+          Queues: [
+             {
+              "Ref": "testeventsrulesqsdeadLetterQueueA4A15A1C",
+            },
+          ]
+    })
+
+})
+
```

Comment:
Add a test case to check for events rule permissions, check [here](https://github.com/awslabs/aws-solutions-constructs/blob/master/source/patterns/%40aws-solutions-constructs/aws-events-rule-lambda/test/events-rule-lambda.test.ts#L142) for reference

Reply:
Included an additional test case for events rule permissions.